code/datums/ai/dog/dog_controller.dm

```diff
@@ -0,0 +1,275 @@
+	if(!isitem(thrown_thing))
+		return
+	if(blackboard[BB_FETCH_IGNORE_LIST][thrown_thing])
+		return
+
+	RegisterSignal(thrown_thing, COMSIG_MOVABLE_THROW_LANDED, .proc/listen_throw_land)
+
+/// A throw we were listening to has finished, see if it's in range for us to try grabbing it
+/datum/ai_controller/dog/proc/listen_throw_land(obj/item/thrown_thing, datum/thrownthing/throwing_datum)
+	SIGNAL_HANDLER
+
+	UnregisterSignal(thrown_thing, list(COMSIG_PARENT_QDELETING, COMSIG_MOVABLE_THROW_LANDED))
+	if(!istype(thrown_thing) || !isturf(thrown_thing.loc) || !can_see(pawn, thrown_thing, length=AI_DOG_VISION_RANGE))
+		return
+
+	current_movement_target = thrown_thing
+	blackboard[BB_FETCH_TARGET] = thrown_thing
+	blackboard[BB_FETCH_DELIVER_TO] = throwing_datum.thrower
+	current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/fetch)
+
+/// Someone's interacting with us by hand, see if they're being nice or mean
+/datum/ai_controller/dog/proc/on_attack_hand(datum/source, mob/living/user)
+	SIGNAL_HANDLER
+
+	if(user.combat_mode)
+		unfriend(user)
+	else
+		if(prob(AI_DOG_PET_FRIEND_PROB))
+			befriend(user)
+		// if the dog has something in their mouth that they're not bringing to someone for whatever reason, have them drop it when pet by a friend
+		var/list/friends = blackboard[BB_DOG_FRIENDS]
+		if(blackboard[BB_SIMPLE_CARRY_ITEM] && !current_movement_target && friends[user])
+			var/obj/item/carried_item = blackboard[BB_SIMPLE_CARRY_ITEM]
+			pawn.visible_message("<span='danger'>[pawn] drops [carried_item] at [user]'s feet!</span>")
+			// maybe have a dedicated proc for dropping things
+			carried_item.forceMove(get_turf(user))
+			blackboard[BB_SIMPLE_CARRY_ITEM] = null
+
+/// Someone is being nice to us, let's make them a friend!
+/datum/ai_controller/dog/proc/befriend(mob/living/new_friend)
+	var/list/friends = blackboard[BB_DOG_FRIENDS]
+	if(friends[new_friend])
+		return
+	if(in_range(pawn, new_friend))
+		new_friend.visible_message("<b>[pawn]</b> licks at [new_friend] in a friendly manner!", "<span class='notice'>[pawn] licks at you in a friendly manner!</span>")
+	friends[new_friend] = TRUE
+	RegisterSignal(new_friend, COMSIG_MOB_POINTED, .proc/check_point)
+	RegisterSignal(new_friend, COMSIG_MOB_SAY, .proc/check_command)
+
+/// Someone is being mean to us, take them off our friends (add actual enemies behavior later)
+/datum/ai_controller/dog/proc/unfriend(mob/living/ex_friend)
+	var/list/friends = blackboard[BB_DOG_FRIENDS]
+	friends[ex_friend] = null
+	UnregisterSignal(ex_friend, list(COMSIG_MOB_POINTED, COMSIG_MOB_SAY))
+
+/// Someone we like is pointing at something, see if it's something we might want to interact with (like if they might want us to fetch something for them)
+/datum/ai_controller/dog/proc/check_point(mob/pointing_friend, atom/movable/pointed_movable)
+	SIGNAL_HANDLER
+
+	if((world.time < blackboard[BB_DOG_COMMAND_CD] + AI_DOG_COMMAND_COOLDOWN)) // command cooldown
```

Comment:
this shouldn't use the blackboard, it can just use normal cooldown macros

Reply:
in general the blackboard only exists if its used in behaviors, because this allows it to be controller-agnostic. that doesn't matter for controller behavior because thats inherently not controller agnostic.

Reply:
in general the blackboard only exists if its used in behaviors, because this allows it to be controller-agnostic. that doesn't matter for controller behavior because thats inherently not controller agnostic.